src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2098,22 +2098,30 @@ def encryption_protector_update(
 #                sql managed instance         #
 ###############################################
 
+class ManagedInstanceTiers(str, Enum):
 
-def _find_managed_instance_sku_from_capabilities(cli_ctx, location, sku):
+    general_purpose = "GeneralPurpose"
+    business_critical = "BusinessCritical"
+
+class ManagedInstanceFamilies(str, Enum):
+
+    gen4 = "Gen4"
+    gen5 = "Gen5"
+
+def _find_managed_instance_sku_from_capabilities(
+        cli_ctx,
+        location,
+        sku,
+        is_instance_create=True):
     '''
     Given a requested sku which may have some properties filled in
-    (e.g. tier and capacity), finds the canonical matching sku
+    (e.g. tier and family), finds the canonical matching sku
     from the given location's capabilities.
     '''
 
     logger.debug('_find_managed_instance_sku_from_capabilities input: %s', sku)
 
-    if sku.name:
-        # User specified sku.name, so nothing else needs to be resolved.
-        logger.debug('_find_managed_instance_sku_from_capabilities return sku as is')
-        return sku
-
-    if not _any_sku_values_specified(sku):
+    if is_instance_create and not _any_sku_values_specified(sku):
```

Comment:
Why is `is_instance_create` logic needed?

Reply:
You are right it is redundant.
The idea was to protect from getting a default sku name  when sku components are not entered during update SLO

However, since SKU for existing instance is populated before this method is called, default sku will not be taken for update SLO.

I'll remove it